RetrofitLab/app/src/main/java/com/ezequielc/retrofitlab/MainActivity.java

```diff
@@ -0,0 +1,95 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mEditText = (EditText) findViewById(R.id.edit_text);
+        mButton = (Button) findViewById(R.id.button);
+        mCityView = (TextView) findViewById(R.id.city);
+        mWeatherView = (TextView) findViewById(R.id.weather);
+        mPressureView = (TextView) findViewById(R.id.pressure);
+        mHumidityView = (TextView) findViewById(R.id.humidity);
+        mTemperatureView = (TextView) findViewById(R.id.temperature);
+
+        mButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String cityname = mEditText.getText().toString();
+                getCityInfo(cityname);
+            }
+        });
+    }
+
+    protected void getCityInfo(String cityname){
+        ConnectivityManager connectivityManager =
+                (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+
+        if (networkInfo != null && networkInfo.isConnected()) {
+            String baseURL = "http://api.openweathermap.org/data/2.5/";
+
+            Retrofit retrofit = new Retrofit.Builder()
+                    .baseUrl(baseURL)
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .build();
+
+            OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
+
+            Call<RootObject> call = service.getInfo(cityname);
+
+            call.enqueue(new Callback<RootObject>() {
+                @Override
+                public void onResponse(Call<RootObject> call, Response<RootObject> response) {
+                    try {
+                        String city = response.body().getName();
+                        String weather = response.body().getWeather().get(0).getDescription();
+                        String pressure = response.body().getMain().getPressure().toString();
+                        String humidity = response.body().getMain().getHumidity().toString();
+                        String temperature = response.body().getMain().getTemp().toString();
+
+                        mCityView.setText("City: " + city);
+                        mWeatherView.setText("Weather: " + weather);
+                        mPressureView.setText("Pressure: " + pressure + " hPa");
+                        mHumidityView.setText("Humidity: " + humidity + " %");
+                        mTemperatureView.setText("Temperature: " + temperature + " °F");
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }
+
+                @Override
+                public void onFailure(Call<RootObject> call, Throwable t) {
+
```

Comment:
You should notify the user (with a Toast or Snackbar) and then call `t.printStackTrace()`